src/server/api/routes/users.router.js

```diff
@@ -111,4 +112,46 @@ router.get('/:id', async (req, res) => {
 // Application routes
 router.use('/:userId/boards', boardsRouter);
 
+/**
+ * @swagger
+ * /users:
+ *  post:
+ *    tags:
+ *    - Users
+ *    summary: Create a user
+ *    description:
+ *      Will create a new user.
+ *    produces: application/json
+ *    parameters:
+ *      - in: body
+ *        name: user
+ *        description: The user to create.
+ *        schema:
+ *          type: object
+ *          required:
+ *            - fullName
+ *            - email
+ *            - firebaseUId
+ *          properties:
+ *            fullName:
+ *              type: string
+ *            email:
+ *              type: string
+ *            firebaseUId:
+ *              type: string
+ *    responses:
+ *      201:
+ *        description: User created
+ *      5XX:
+ *        description: Unexpected error.
+ *      400:
+ *        description: Bad request.
+ *      404:
+ *        description: Not found.
+ */
+router.post('/', [authenticate], async (req, res) => {
```

Comment:
Good, but we need to check if the user is already created. And we always want to return the user. The userid will then be stored in localstorage in the frontend

Reply:
I checked it by post duplicate email or name and it didn't save data if the email or name is already in the database. Also if authenticate also work

Reply:
I tested it with both postman and swagger. 
for duplicate data, I received this message  in swagger 
![image](https://user-images.githubusercontent.com/75579156/129729715-10f78c3f-975f-4e41-b5a1-8532e69bc8f0.png)

and in postman 
![image](https://user-images.githubusercontent.com/75579156/129729789-992a17f3-4375-4306-8338-5a9ca4e66b6f.png)

if I used to authenticate then this message 
![image](https://user-images.githubusercontent.com/75579156/129729971-d25ab579-1755-42c4-92f3-f59e1221b2a4.png)

Reply:
I think if your knex query throws then you won't be able to return the user. You will have to check if the user exists first. I assume you can reuse the controller method to get the user

Reply:
I didn't understand it. how can do that?

Reply:
You will receive the fiebase user uid in the body. 
Then create a knex method to check if a user exists with that uid. If so, return that user. Otherwise create a new user and return that.